backend/api/Services/MissionSchedulingService.cs

```diff
@@ -53,37 +53,36 @@ public async Task StartNextMissionRunIfSystemIsAvailable(string robotId)
             if (missionRun == null)
             {
                 logger.LogInformation("The robot was ready to start mission, but no mission is scheduled");
-                if (!robot.MissionQueueFrozen)
+                if (robot.MissionQueueFrozen) { return; }
+
+                try { missionRun = await returnToHomeService.ScheduleReturnToHomeMissionRunIfRobotIsNotHome(robot.Id); }
+                catch (ReturnToHomeMissionFailedToScheduleException)
                 {
-                    try { await returnToHomeService.ScheduleReturnToHomeMissionRunIfRobotIsNotHome(robot.Id); }
-                    catch (ReturnToHomeMissionFailedToScheduleException)
-                    {
-                        logger.LogError("Failed to schedule a return to home mission for robot {RobotId}", robot.Id);
-                        await robotService.UpdateCurrentArea(robot.Id, null);
-                    }
+                    logger.LogError("Failed to schedule a return to home mission for robot {RobotId}", robot.Id);
+                    await robotService.UpdateCurrentArea(robot.Id, null);
                 }
-                return;
+                if (missionRun == null) { return; }  // The robot is already home
             }
 
-            if (!await TheSystemIsAvailableToRunAMission(missionRun.Robot.Id, missionRun.Id))
+            if (!await TheSystemIsAvailableToRunAMission(robotId, missionRun.Id))
             {
                 logger.LogInformation("Mission {MissionRunId} was put on the queue as the system may not start a mission now", missionRun.Id);
                 return;
             }
 
 
             // Verify that localization is fine
-            if (!await localizationService.RobotIsLocalized(missionRun.Robot.Id) && !missionRun.IsLocalizationMission())
+            if (!await localizationService.RobotIsLocalized(robotId) && !missionRun.IsLocalizationMission())
             {
                 logger.LogError("Tried to schedule mission {MissionRunId} on robot {RobotId} before the robot was localized, scheduled missions will be canceled", missionRun.Id, missionRun.Robot.Id);
                 try { await CancelAllScheduledMissions(robot.Id); }
                 catch (RobotNotFoundException) { logger.LogError("Failed to cancel scheduled missions for robot {RobotId}", robot.Id); }
                 return;
             }
 
-            if (!missionRun.IsLocalizationMission() && !await localizationService.RobotIsOnSameDeckAsMission(missionRun.Robot.Id, missionRun.Area.Id))
+            if (!missionRun.IsLocalizationMission() && !await localizationService.RobotIsOnSameDeckAsMission(robot.Id, missionRun.Area.Id))
```

Comment:
Should `robotId` been used instead of `robot.Id`? For consistency, look over the whole function